File: ObjectiveC/CASSample/AdDelegate.m
//
//  AdDelegate.m
//  CASSample
//

#import "AdDelegate.h"

@implementation AdDelegate

- (id)initWithType:(CASType)type {
    self = [super init];
    if (self) {
        self.type = type;
    }
    return self;
}

+ (NSString *)getNameOfType:(CASType)type{
    switch (type) {
        case CASTypeBanner:
            return @"Banner";
        case CASTypeInterstitial:
            return @"Interstitial";
        case CASTypeRewarded:
            return @"Rewarded";
        case CASTypeNative:
            return @"Native";
        default:
            return @"None";
    }
}

- (NSString *)getTypeName {
    return [AdDelegate getNameOfType:self.type];
}

- (void)willShownWithAd:(id<CASStatusHandler>)adStatus {
    if (_lastInfo) {
        [_lastInfo setText:adStatus.identifier];
        NSLog(@"[CAS Sample] %@d received Show action", [self getTypeName]);
    }
}

- (void)didShowAdFailedWithError:(NSString *)error {
    NSLog(@"[CAS Sample] %@d Ad show failed: %@d", [self getTypeName], error);
}

- (void)didClickedAd {
    NSLog(@"[CAS Sample] %@d Ad received Click action", [self getTypeName]);
}

- (void)didCompletedAd {
    NSLog(@"[CAS Sample] %@d Ad complete. You have been rewarded", [self getTypeName]);
}

- (void)didClosedAd {
    NSLog(@"[CAS Sample] %@d Ad received Close action", [self getTypeName]);
}

@end

File: Swift/CASSample/AdDelegate.swift
//
//  AdDelegate.swift
//  CASSample
//
//  Copyright © 2020 Clever Ads Solutions. All rights reserved.
//

import CleverAdsSolutions
import Foundation
import UIKit

class AdDelegate: CASCallback {
    let type: CASType
    var lastInfo:UILabel?

    init(type: CASType) {
        self.type = type
    }

    func willShown(ad adStatus: CASStatusHandler) {
        lastInfo?.text = adStatus.identifier
        print("[CAS Sample] \(type.description) Ad received Show action")
    }

    func didShowAdFailed(error: String) {
        print("[CAS Sample] \(type.description) Ad show failed: \(error)")
    }

    func didClickedAd() {
        print("[CAS Sample] \(type.description) Ad received Click action")
    }

    func didCompletedAd() {
        print("[CAS Sample] \(type.description) Ad complete. You have been rewarded")
    }

    func didClosedAd() {
        print("[CAS Sample] \(type.description) Ad received Close action")
    }
}
